mbedtls/src/ssl/context.rs

```diff
@@ -301,42 +336,44 @@ pub struct HandshakeContext<'ctx> {
 // - mbedtls not providing any callbacks on handshake finish.
 // - no reasonable way to obtain a storage within the sni callback tied to the handshake or to the rust Context. (without resorting to a unscalable map or pointer magic that mbedtls may invalidate)
 //
-impl<'ctx> HandshakeContext<'ctx> {
-
-    pub(crate) fn init(context: &'ctx mut Context) -> Self {
-        HandshakeContext { context }
+impl HandshakeContext {
+    pub fn reset_handshake(&mut self) {
+        self.handshake_cert.clear();
+        self.handshake_pk.clear();
+        self.handshake_ca_cert = None;
+        self.handshake_crl = None;
     }
     
     pub fn set_authmode(&mut self, am: AuthMode) -> Result<()> {
-        if self.context.inner.handshake as *const _ == ::core::ptr::null() {
+        if self.inner.handshake as *const _ == ::core::ptr::null() {
             return Err(Error::SslBadInputData);
         }
         
-        unsafe { ssl_set_hs_authmode(self.context.into(), am as i32) }
+        unsafe { ssl_set_hs_authmode(self.into(), am as i32) }
         Ok(())
     }
 
     pub fn set_ca_list(
         &mut self,
-        chain: Arc<MbedtlsList<Certificate>>,
+        chain: Option<Arc<MbedtlsList<Certificate>>>,
```

Comment:
This change doesn't make sense to me. Is the intention to set only the CRL? The behavior when passing None should be clearly documented.

Reply:
It allows unsetting both the crl and ca chain (setting them to null)

This helps if they were previously set and they need to be altered.